### PR TITLE
dotnet-sdk-sts: Add version 7.0.100

### PR DESCRIPTION
**Pull request by alexchapar**
## What type of Apps related PR is this? (check all applicable)
- [x] ✨ Add Manifest
- [ ] ♻️ Refactor
- [ ] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 🚩 Other

## Description
dotnet-sdk-sts is for the "Standard Term Support" version of the dotnet SDK. It is installed side-by-side and "override" the DOTNET_ROOT variable to be able to use it directly. We can always switch easily between the dotnet-sdk (LTS) and this one by using the `scoop reset dotnet-sdk` or `scoop reset dotnet-sdk-sts`.

The use of `dotnet --version` can also be helpful to know which one is actually active

## Related Tickets & Documents
None

## [optional] What gif/emoji best describes this PR or how it makes you feel?
:tada: :dancers: :sparkles: :zap:

### Add/Modify a description in the manifest?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Check URLs?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Update to the latest version?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Update Hashes?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Format the json manifest?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Notes in the manifest?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed

### Application Dependencies?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed

### Application Suggestions?
- [x] 👍 yes
- [ ] 🙋 yes, but I need help
- [ ] 🙅 no, because they aren't needed

### Pre-installation steps to perform?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed

### Post-installation steps to perform?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed

### File persistence to perform?
- [ ] 👍 yes
- [ ] 🙋 yes, but I need help
- [x] 🙅 no, because they aren't needed

### Other ?
- [x] :rotating_light: Change the Environment variable required (this will impact the dotnet-sdk version soon)
- [x] Added the arm64
- [x] Changes the regex (from the lts version) to obtain the version so it wil remove risks of getting any "sts" version (aka "eol", "maintenance", etc.)

### CODE REVIEW

**Comment by BotServiceAutomation** [comment]
All changes look good.

Wait for review from human collaborators.
### dotnet-sdk-sts

- [x] Description
- [x] License
- [x] Hashes
- [x] Checkver
- [x] Autoupdate
- [x] Autoupdate Hash Extraction

**Review by Bibz87 — APPROVED** [review]

**Pull request merged**
